GVFS/GVFS.Common/Prefetch/Jobs/ReadFilesJob.cs

```diff
@@ -44,11 +44,24 @@ protected override void DoWork()
                     foreach (string path in this.blobIdToPaths[blobId])
                     {
                         bool succeeded = false;
-                        using (SafeFileHandle handle = NativeFileReader.Open(path))
+
+                        // TODO(Mac): Replace this with a native method as opposed to a slow .NET method
+                        if (!GVFSPlatform.Instance.IsUnderConstruction)
+                        {
+                            using (SafeFileHandle handle = NativeFileReader.Open(path))
+                            {
+                                if (!handle.IsInvalid)
+                                {
+                                    succeeded = NativeFileReader.ReadOneByte(handle, buffer);
+                                }
+                            }
+                        }
+                        else
                         {
-                            if (!handle.IsInvalid)
+                            using (FileStream filestream = new FileStream(path, FileMode.Open, FileAccess.Read))

```

Comment:
That said, this code can throw IOExceptions, and we need to catch those and report them as errors on the individual files, not fail the whole prefetch operation.